src/components/AvatarGroup/AvatarGroupHelper.ts

```diff
@@ -0,0 +1,6 @@
+import React from "react";
+import { AvatarProps } from "../Avatar/Avatar";
+
+export const avatarOnClick = (event: React.MouseEvent | React.KeyboardEvent, avatarProps: AvatarProps) => {
+  return avatarProps?.onClick && (() => avatarProps.onClick(event, avatarProps?.id));
```

Comment:
it changes the signature a bit, but I think it's more readable. WDYT?
```suggestion
  return () => avatarProps?.onClick?.(event, avatarProps.id);
```